### PR TITLE
Debounce manual advancements after auto advancements in stories

### LINKED ISSUE #18564: Debounce amp-story navigation events
It's possible that the user could tap immediately before or after an auto-advance navigation, and the result is that the story will navigate twice.  We should debounce these events within a reasonable amount of time, to ensure that only one navigation happens in this case.

It's probably safe to NOT debounce two auto advancements, or two manual advancements; we should only debounce in the case of one auto advancement and one manual advancement.

### PR DESCRIPTION
**Pull request by newmuis**
Fixes #18564

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story.js [hunk 1]
```diff
@@ -1343,6 +1357,30 @@ export class AmpStory extends AMP.BaseElement {
...
+    if (previousAdvancement) {
+      const elapsedMsSinceLastNavigation =
+        currentAdvancement.timestamp - previousAdvancement.timestamp;
+
+      if (
+        elapsedMsSinceLastNavigation < NAVIGATION_DEBOUNCE_THRESHOLD_MS &&
+        isAutomaticAdvancement(previousAdvancement.mode) &&
+        isManualAdvancement(currentAdvancement.mode)
+      ) {
+        return;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story.js]: Nit: `return Promise.reject();`

File: extensions/amp-story/1.0/amp-story.js [hunk 2]
```diff
@@ -165,6 +171,14 @@ const Attributes = {
  */
 const PAGE_LOAD_TIMEOUT_MS = 5000;
 
+/**
+ * The duration of time (in milliseconds) after a story has automatically
+ * advanced during which the user will not be able to manually advance, to
+ * prevent accidental duplicate advances.
+ * @private @const {number}
+ */
+const NAVIGATION_DEBOUNCE_THRESHOLD_MS = 750;
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story.js]: Personal opinion backed by not much: given that other stories platform like Instagram/TikTok/Snapchat don't do anything here and let users solve this on their own, I think this threshold should be a lot lower.
There is some consensus on the web around a magic ~300ms number, that has been used for web perf recommendations (all actions should be under this) or by browsers (timer the browser used to wait to see if a tap could become a double tap).
Up to you though

Comment (newmuis) [reply extensions/amp-story/1.0/amp-story.js]: 300ms works for me.  I had it as that initially and lengthened it based on just feeling it out, but I think what you say makes sense.  Of course, we can always adjust this number over time if it's really that big of an issue, but erring on the side of a short threshold sounds like the right principle here.

**Comment by newmuis** [comment]
This is waiting for the rollforward of the story analytics changes, which include a refactoring on which this PR is dependent.

**Comment by Enriqe** [comment]
SG, this should be ready shortly after #24548

**Comment by mrjoro** [comment]
Is this ready to go in yet? :)

**Comment by gmajoulet** [comment]
We still want to do this, but the code is unfortunately obsolete. We do have ways to track the previous advancement etc now. I tagged the issue into our new issues tracking dashboard so we can get to it. Jon if you want to write code feel free to re-open this PR :))

**Pull request closed without merging**